ggdeploymentd/src/component_config.c

```diff
@@ -46,13 +49,46 @@ static GglError apply_reset_config(
                      "update not provided as a buffer.");
             return GGL_ERR_INVALID;
         }
-        ret = ggl_gg_config_delete(GGL_BUF_LIST(
-            GGL_STR("services"),
-            component_name,
-            GGL_STR("configuration"),
-            reset_element->buf
-        ));
 
+        // Empty string means they want to reset the whole configuration to
+        // default configuration.
+        if (ggl_buffer_eq(reset_element->buf, GGL_STR(""))) {
+            GGL_LOGI(
+                "Received a request to reset the entire configuration for %.*s",
+                (int) component_name.len,
+                component_name.data
+            );
+            ret = ggl_gg_config_delete(GGL_BUF_LIST(
+                GGL_STR("services"), component_name, GGL_STR("configuration")
+            ));
+            if (ret != GGL_ERR_OK) {
+                GGL_LOGE(
+                    "Error while deleting the component %.*s's configuration.",
+                    (int) component_name.len,
+                    component_name.data
+                );
+            }
+
+            break;
```

Comment:
I think this should fail the deployment? Because if delete errors then either there was something wrong with their request they would want to know about, or something really bad happened in the config which they would probably also want to know about because the config state will not be as expected after that

Reply:
Meant to put this on line 64